### PR TITLE
[1LP][WIPTEST] Appliance SSH use and SystemD service updates

### PR DESCRIPTION
**Pull request by mshriver**
Do some cleanup in the appliance method, two main goals:
1. Remove direct systemctl calls and use SystemdService objects
2.  Use ssh in a context manager

### CODE REVIEW

**Comment by jawatts** [comment]
PRT results don't look great, can you review them please.

File: cfme/utils/appliance/__init__.py [hunk 1]
```diff
@@ -522,13 +528,14 @@ def configure(self, log_callback=None, **kwargs):
...
-            self.ssh_client.run_command("systemctl daemon-reload", ensure_host=True)
-            self.ssh_client.run_command("service auditd restart", ensure_host=True)
+            with self.ssh_client as ssh_client:
+                ssh_client.run_command(
+                    "cp -pr /etc/sysconfig/network-scripts/ifcfg-eth0 /var/tmp", ensure_host=True)
+                ssh_client.run_command(
+                    "echo '-w /etc/sysconfig/network-scripts/ifcfg-eth0 -p wa' >> "
+                    "/etc/audit/rules.d/audit.rules", ensure_host=True)
+                self.httpd.daemon_reload()
+                self.auditd.restart()
```
Comment (mshriver) [inline cfme/utils/appliance/__init__.py]: auditd can't be start/stop via systemctl, RH directs users to use service command for auditd restart.

File: cfme/utils/appliance/__init__.py [hunk 2]
```diff
@@ -985,10 +993,11 @@ def diagnose_evm_failure(self):
...
             # postgres isn't running, try to start it
-            cmd = 'systemctl restart {}-postgresql'.format(self.db.postgres_version)
-            result = self.db.ssh_client.run_command(cmd)
-            if result.failed:
-                return 'postgres failed to start:\n{}'.format(result.output)
+            logger.info('Database is not online, restarting')
+            try:
+                self.db_service.restart()
+            except SystemdException:
+                return 'postgres failed to start'
```
Comment (izapolsk) [inline cfme/utils/appliance/__init__.py]: Do you thing result isn't necessary in this case ?

File: cfme/utils/appliance/__init__.py [hunk 3]
```diff
@@ -1455,41 +1469,23 @@ def is_web_ui_running(self, unsure=False):
...
-        if expected_exit_code is not None and result.rc != expected_exit_code:
-            msg = ('Failed to {} evmserverd on {}\nError: {}'
-                   .format(command, self.hostname, result.output))
-            log_callback(msg)
-            raise ApplianceException(msg)
-
-        return result.rc
-
     @logger_wrap("Status of EVM service: {}")
     def is_evm_service_running(self, log_callback=None):
```
Comment (izapolsk) [inline cfme/utils/appliance/__init__.py]: I'd also get rid of those methods as well in favor of service calls.

**Comment by izapolsk** [comment]
Looks good so far. but please check PRT errors related to evmserverd. maybe BZ.

File: cfme/utils/appliance/__init__.py (hunk 3 above)
Comment (mshriver) [reply cfme/utils/appliance/__init__.py]: done in latest commit

File: cfme/utils/appliance/__init__.py (hunk 2 above)
Comment (mshriver) [reply cfme/utils/appliance/__init__.py]: included logging in latest commit

**Pull request merged**